File: split/trapzeroglue.pas
procedure trapzeroglue;
begin
  if (mem[curval + 1].int = 0) and (mem[curval + 2].int = 0) and (mem[curval + 3].int = 0) then
  begin
    mem[0].hh.rh := mem[0].hh.rh + 1;
    deleteglueref(curval);
    curval := 0;
  end;
end;

File: cpp/src/trapzeroglue.cpp
#include "trapzeroglue.h"
#include "noeud.h"
#include "deleteglueref.h"

GlueSpec *trapzeroglue(GlueSpec *g)
{
	if (g->width == 0 && g->stretch == 0 && g->shrink == 0)
	{
		zero_glue->glue_ref_count++;
		deleteglueref(g);
		return nullptr;
	}
	return g;
}
